R/try.R

```diff
@@ -13,24 +13,24 @@
 #' FunDoesNotExist(1:10) %except% MEAN(1:10) %except% mean(1:10)
 #' FunDoesNotExist(1:10) %except% (MEAN(1:10) %except% mean(1:10))
 `%except%` <- function(try, except) {
-
-    call <- sys.call(-1)
-    env <- parent.frame()
-    try <- substitute(try)
-    fallback <- substitute(except)
+    # Need to capture these in the evaluation frame of `%except%` but only want
+    # to do the work if there's an error
+    delayedAssign("call", sys.call(-1))
+    delayedAssign("env", parent.frame())
+    delayedAssign("except_text", deparse(substitute(except)))
+    delayedAssign("try_text", deparse(substitute(try)))
 
     tryCatch(
-        eval(try, envir = env),
+        try,
         error = function(e) {
             log_level(
-                WARN,
-                paste(
-                    'Running', shQuote(deparse(fallback)), 'as',
-                    shQuote(deparse(try)), 'failed:',
-                    shQuote(e$message)),
-                namespace = 'except',
-                .topcall = call, .topenv = env)
-            eval(fallback, envir = env)
+              WARN,
+              paste0('Running \'', except_text, '\' as \'', try_text, '\' failed: \'', e$message, '\''),
```

Comment:
I converted `shQuote()` to `'` because `shQuote()` is really for quoting shell arguments, and uses a different quoting character on different platforms, leading to spurious diffs.